recipe.cake

```diff
@@ -132,6 +167,111 @@ Task("Prepare-Chocolatey-Packages")
     }
 });
 
+Task("Build-ChocolateyNo7zip")
+    .WithCriteria(() => BuildParameters.Configuration == "ReleaseOfficial", "Skipping No7zip because this isn't an official release")
+    .IsDependentOn("Build")
+    .IsDependentOn("Test")
+    .IsDependeeOf("Run-ILMerge")
+    .Does<BuildData>(data => RequireTool(ToolSettings.MSBuildExtensionPackTool, () =>
+{
+    Information("Building {0} with No7zip", BuildParameters.SolutionFilePath);
+
+    CleanDirectory(BuildParameters.Paths.Directories.PublishedApplications + "/choco-no7zip/");
+
+    var no7zLogPath = BuildParameters.Paths.Files.BuildLogFilePath.ToString().Replace("\\.(\\S+)$", "-no7zip.${1}");
+
+    if (BuildParameters.BuildAgentOperatingSystem == PlatformFamily.Windows)
+    {
+        var msbuildSettings = new MSBuildSettings()
+            .SetPlatformTarget(ToolSettings.BuildPlatformTarget)
+            .UseToolVersion(ToolSettings.BuildMSBuildToolVersion)
+            .WithProperty("OutputPath", MakeAbsolute(new DirectoryPath(BuildParameters.Paths.Directories.PublishedApplications + "/choco-no7zip/")).FullPath)
+            .WithProperty("TreatWarningsAsErrors", BuildParameters.TreatWarningsAsErrors.ToString())
+            .WithTarget("Build")
+            .SetMaxCpuCount(ToolSettings.MaxCpuCount)
+            .SetConfiguration("ReleaseOfficialNo7zip")
+            .WithLogger(
+                Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
+                "XmlFileLogger",
+                string.Format(
+                    "logfile=\"{0}\";invalidCharReplacement=_;verbosity=Detailed;encoding=UTF-8",
+                    no7zLogPath
+                )
+            );
+
+        MSBuild(BuildParameters.SolutionFilePath, msbuildSettings);
+    }
+
+    if (FileExists(no7zLogPath))
+    {
+        BuildParameters.BuildProvider.UploadArtifact(no7zLogPath);
+    }
+}));
+
+Task("Prepare-ChocolateyNo7zip-Package")
+    .WithCriteria(() => BuildParameters.Configuration == "ReleaseOfficial", "Skipping No7zip because this isn't an official release")
```

Comment:
Originally I didn't have the criteria on prepare or pack, but my dependecy-fu was lacking. If there's a better way to do this, I'm very happy to make changes!

Reply:
I have had a stab at making these consistent, so going to resolve this.